[glsl-out] Operands to "%" must be integral

### Example:
This valid glsl:
```glsl
float out = mod(1.0,1.0);
```
Translates to this wgls
```rs
let out: f32 = 1.0 % 1.0;
```
And this wgls translates to this glsl:
```glsl
float out = 1.0 % 1.0;
```
The resulting shader does not work.

When run with wgpu it outputs:
`Internal error in VERTEX shader: 0(27) : error C1021: operands to "%" must be integral`

--- a/src/back/glsl/mod.rs
+++ b/src/back/glsl/mod.rs
@@ -2117,13 +2117,13 @@ impl<'a, W: Write> Writer<'a, W> {
             Expression::Binary { op, left, right } => {
                 // Holds `Some(function_name)` if the binary operation is
                 // implemented as a function call
-                use crate::BinaryOperator as Bo;
+                use crate::{BinaryOperator as Bo, ScalarKind as Sk, TypeInner as Ti};
 
-                let function = if let (&TypeInner::Vector { .. }, &TypeInner::Vector { .. }) = (
-                    ctx.info[left].ty.inner_with(&self.module.types),
-                    ctx.info[right].ty.inner_with(&self.module.types),
-                ) {
-                    match op {
+                let left_inner = ctx.info[left].ty.inner_with(&self.module.types);
+                let right_inner = ctx.info[right].ty.inner_with(&self.module.types);
+
+                let function = match (left_inner, right_inner) {
+                    (&Ti::Vector { .. }, &Ti::Vector { .. }) => match op {
                         Bo::Less => Some("lessThan"),
                         Bo::LessEqual => Some("lessThanEqual"),
                         Bo::Greater => Some("greaterThan"),
@@ -2131,9 +2131,14 @@ impl<'a, W: Write> Writer<'a, W> {
                         Bo::Equal => Some("equal"),
                         Bo::NotEqual => Some("notEqual"),
                         _ => None,
-                    }
-                } else {
-                    None
+                    },
+                    _ => match (left_inner.scalar_kind(), right_inner.scalar_kind()) {
+                        (Some(Sk::Float), _) | (_, Some(Sk::Float)) => match op {
+                            Bo::Modulo => Some("mod"),
+                            _ => None,
+                        },
+                        _ => None,
+                    },
                 };
 
                 write!(self.out, "{}(", function.unwrap_or(""))?;

